### ISSUE TITLE
`g:UltiSnipsEnableSnipMate` won't work if in plugin directory

### ISSUE THREAD

**Issue by g6ai** [SirVer/ultisnips#1424]
**Expected behavior:**
`g:UltiSnipsEnableSnipMate` to work if it is set in a file in the runtime's plugin directory.

**Actual behavior:**
It only works if it's placed in `init.vim`.

-----

- **Operating System**: macOS 10.14.6
- **Vim Version**: Neovim 0.6.0
- **UltiSnips Version**: 3.2
- **Python inside Vim**: 3.8.10
- **Docker repo/vimrc**: https://github.com/g6ai/dotfiles/blob/5d1be0b295df4339eab0275a26e8190b5a303041/.chezmoitemplates/vim/plugin/ultisnips.vim#L6

**Comment by SirVer** [comment]
@g6ai UltiSnips reads the variable when it parses a snippet, which happens the first time you expand one. It does not care where the variable is set. 

Could it be that the problem here is be the order in which Vim executes the `.vim` files. Could you add the order of scripts that are loaded, see https://vim.fandom.com/wiki/List_loaded_scripts.

Also thanks for providing a docker repo, however I do not know what keys I have to press to reproduce your problem and what you would expect to happen instead of what you see.

**Comment by g6ai** [comment]
Thank you @SirVer for the info! Here are the order of scripts loaded, for a `.tex` LaTeX buffer:

<details>
<p>

```
  1: ~/.config/nvim/init.vim
  2: ~/.config/nvim/core/misc.vim
  3: ~/.config/nvim/pack/plugins/opt/chezmoi.vim/filetype.vim
  4: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/filetype.vim
  5: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/indent.vim
  6: ~/.config/nvim/core/cursor.vim
  7: ~/.config/nvim/core/spell.vim
  8: ~/.config/nvim/core/colour.vim
  9: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/syntax/syntax.vim
 10: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/syntax/synload.vim
 11: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/colors/slate.vim
 12: ~/.config/nvim/core/clipboard.vim
 13: ~/.config/nvim/core/buffer.vim
 14: ~/.config/nvim/core/window.vim
 15: ~/.config/nvim/core/python.vim
 16: ~/.config/nvim/core/plug.vim
 17: ~/.config/nvim/autoload/plug.vim
 18: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/ftoff.vim
 19: ~/.config/nvim/plugged/vim-fugitive/ftdetect/fugitive.vim
 20: ~/.config/nvim/plugged/vimtex/ftdetect/cls.vim
 21: ~/.config/nvim/plugged/vimtex/ftdetect/tex.vim
 22: ~/.config/nvim/plugged/vimtex/ftdetect/tikz.vim
 23: ~/.config/nvim/plugged/vim-markdown/ftdetect/markdown.vim
 24: ~/.config/nvim/plugged/ultisnips/ftdetect/snippets.vim
 25: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/fusion.vim
 26: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/gdresource.vim
 27: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/gdscript.vim
 28: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/glimmer.vim
 29: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/glsl.vim
 30: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/gowork.vim
 31: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/graphql.vim
 32: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/hack.vim
 33: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/hcl.vim
 34: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/heex.vim
 35: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/hjson.vim
 36: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/json5.vim
 37: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/ledger.vim
 38: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/nix.vim
 39: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/prisma.vim
 40: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/pug.vim
 41: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/ql.vim
 42: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/query.vim
 43: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/surface.vim
 44: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/teal.vim
 45: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/tlaplus.vim
 46: ~/.config/nvim/plugged/nvim-treesitter/ftdetect/yang.vim
 47: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/ftplugin.vim
 48: ~/.config/nvim/core/gruvbox-material.vim
 49: ~/.config/nvim/core/theme.vim
 50: ~/.config/nvim/autoload/theme.vim
 51: ~/.config/nvim/plugged/gruvbox-material/colors/gruvbox-material.vim
 52: ~/.config/nvim/plugged/gruvbox-material/autoload/gruvbox_material.vim
 53: ~/.config/nvim/plugged/gruvbox-material/after/ftplugin/text/gruvbox_material.vim
 54: ~/Documents/TeX/phd_thesis/.nvimrc
 55: ~/Documents/TeX/phd_thesis/.vimrc
 56: ~/.config/nvim/plugin/better-escape.vim
 57: ~/.config/nvim/plugin/coc.vim
 58: ~/.config/nvim/plugin/doge.vim
 59: ~/.config/nvim/plugin/flog.vim
 60: ~/.config/nvim/plugin/fugitive.vim
 61: ~/.config/nvim/plugin/indent-guides.vim
 62: ~/.config/nvim/plugin/lightline.vim
 63: ~/.config/nvim/plugin/lualine.vim
 64: ~/.config/nvim/plugin/markdown-preview.vim
 65: ~/.config/nvim/plugin/nerdtree.vim
 66: ~/.config/nvim/plugin/nvim-colorizer.vim
 67: ~/.config/nvim/plugin/nvim-telescope.vim
 68: ~/.config/nvim/plugin/nvim-treesitter.vim
 69: ~/.config/nvim/plugin/tmuxline.vim
 70: ~/.config/nvim/plugin/ultisnips.vim
 71: ~/.config/nvim/plugin/vimtex.vim
 72: ~/.config/nvim/plugged/vim-fugitive/plugin/fugitive.vim
 73: ~/.config/nvim/plugged/vim-flog/plugin/flog.vim
 74: ~/.config/nvim/plugged/vim-indent-guides/plugin/indent_guides.vim
 75: ~/.config/nvim/plugged/vim-indent-guides/autoload/indent_guides.vim
 76: ~/.config/nvim/plugged/tmuxline.vim/plugin/tmuxline.vim
 77: ~/.config/nvim/plugged/vimtex/plugin/vimtex.vim
 78: ~/.config/nvim/plugged/tabular/plugin/Tabular.vim
 79: ~/.config/nvim/plugged/ultisnips/plugin/UltiSnips.vim
 80: ~/.config/nvim/plugged/ultisnips/autoload/UltiSnips/map_keys.vim
 81: ~/.config/nvim/plugged/vim-snippets/plugin/vimsnippets.vim
 82: ~/.config/nvim/plugged/vim-smoothie/plugin/smoothie.vim
 83: ~/.config/nvim/plugged/vim-doge/plugin/doge.vim
 84: ~/.config/nvim/plugged/vim-doge/autoload/doge/helpers.vim
 85: ~/.config/nvim/plugged/vim-devicons/plugin/webdevicons.vim
 86: ~/.config/nvim/plugged/committia.vim/plugin/committia.vim
 87: ~/.config/nvim/plugged/vim-startuptime/plugin/startuptime.vim
 88: ~/.config/nvim/plugged/coc.nvim/plugin/coc.vim
 89: ~/.config/nvim/plugged/coc.nvim/autoload/coc/rpc.vim
 90: ~/.config/nvim/plugged/coc.nvim/autoload/coc/util.vim
 91: ~/.config/nvim/plugged/coc.nvim/autoload/coc/client.vim
 92: ~/.config/nvim/plugged/nvim-colorizer.lua/plugin/colorizer.vim
 93: ~/.config/nvim/plugged/plenary.nvim/plugin/plenary.vim
 94: ~/.config/nvim/plugged/nvim-treesitter/plugin/nvim-treesitter.vim
 95: ~/.config/nvim/plugged/nvim-web-devicons/plugin/nvim-web-devicons.vim
 96: ~/.config/nvim/plugged/telescope.nvim/plugin/telescope.vim
 97: ~/.config/nvim/plugged/nvim-hlslens/plugin/hlslens.vim
 98: ~/.config/nvim/plugged/nvim-bqf/plugin/bqf.vim
 99: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/gzip.vim
100: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/health.vim
101: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/man.vim
102: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/matchit.vim
103: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/pack/dist/opt/matchit/plugin/matchit.vim
104: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/matchparen.vim
105: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/netrwPlugin.vim
106: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/rplugin.vim
107: /nix/store/vq40qwg3awmhdw0jas2l3d4wxdg3xy0v-neovim-0.6.0/rplugin.vim
108: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/shada.vim
109: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/spellfile.vim
110: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/tarPlugin.vim
111: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/tohtml.vim
112: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/tutor.vim
113: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/plugin/zipPlugin.vim
114: ~/.config/nvim/plugged/tabular/after/plugin/TabularMaps.vim
115: ~/.config/nvim/plugged/tabular/autoload/tabular.vim
116: ~/.config/nvim/plugged/ultisnips/after/plugin/UltiSnips_after.vim
117: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/autoload/dist/ft.vim
118: ~/.config/nvim/plugged/vimtex/indent/tex.vim
119: ~/.config/nvim/plugged/vimtex/autoload/vimtex/options.vim
120: ~/.config/nvim/plugged/vimtex/autoload/vimtex/util.vim
121: ~/.config/nvim/plugged/vimtex/autoload/vimtex/re.vim
122: ~/.config/nvim/plugged/vimtex/autoload/vimtex/delim.vim
123: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/indent/tex.vim
124: ~/.config/nvim/plugged/vimtex/syntax/tex.vim
125: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/core.vim
126: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/nested.vim
127: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/syntax/tex.vim
128: ~/.config/nvim/plugged/vimtex/ftplugin/tex.vim
129: ~/.config/nvim/plugged/vimtex/autoload/vimtex.vim
130: ~/.config/nvim/plugged/vimtex/autoload/vimtex/state.vim
131: ~/.config/nvim/plugged/vimtex/autoload/vimtex/parser.vim
132: ~/.config/nvim/plugged/vimtex/autoload/vimtex/parser/tex.vim
133: ~/.config/nvim/plugged/vimtex/autoload/vimtex/paths.vim
134: ~/.config/nvim/plugged/vimtex/autoload/vimtex/state/class.vim
135: ~/.config/nvim/plugged/vimtex/autoload/vimtex/cache.vim
136: ~/.config/nvim/plugged/vimtex/autoload/vimtex/parser/fls.vim
137: ~/.config/nvim/plugged/vimtex/autoload/vimtex/view.vim
138: ~/.config/nvim/plugged/vimtex/autoload/vimtex/view/skim.vim
139: ~/.config/nvim/plugged/vimtex/autoload/vimtex/view/_template.vim
140: ~/.config/nvim/plugged/vimtex/autoload/vimtex/compiler.vim
141: ~/.config/nvim/plugged/vimtex/autoload/vimtex/compiler/latexmk.vim
142: ~/.config/nvim/plugged/vimtex/autoload/vimtex/compiler/_template.vim
143: ~/.config/nvim/plugged/vimtex/autoload/vimtex/qf.vim
144: ~/.config/nvim/plugged/vimtex/autoload/vimtex/qf/latexlog.vim
145: ~/.config/nvim/plugged/vimtex/autoload/vimtex/toc.vim
146: ~/.config/nvim/plugged/vimtex/autoload/vimtex/fold.vim
147: ~/.config/nvim/plugged/vimtex/autoload/vimtex/fold/envs.vim
148: ~/.config/nvim/plugged/vimtex/autoload/vimtex/fold/cmd_single_opt.vim
149: ~/.config/nvim/plugged/vimtex/autoload/vimtex/fold/markers.vim
150: ~/.config/nvim/plugged/vimtex/autoload/vimtex/fold/preamble.vim
151: ~/.config/nvim/plugged/vimtex/autoload/vimtex/fold/sections.vim
152: ~/.config/nvim/plugged/vimtex/autoload/vimtex/fold/cmd_addplot.vim
153: ~/.config/nvim/plugged/vimtex/autoload/vimtex/fold/env_options.vim
154: ~/.config/nvim/plugged/vimtex/autoload/vimtex/fold/cmd_multi.vim
155: ~/.config/nvim/plugged/vimtex/autoload/vimtex/fold/items.vim
156: ~/.config/nvim/plugged/vimtex/autoload/vimtex/fold/cmd_single.vim
157: ~/.config/nvim/plugged/vimtex/autoload/vimtex/context.vim
158: ~/.config/nvim/plugged/vimtex/autoload/vimtex/context/cite.vim
159: ~/.config/nvim/plugged/vimtex/autoload/vimtex/bib.vim
160: ~/.config/nvim/plugged/vimtex/autoload/vimtex/cmd.vim
161: ~/.config/nvim/plugged/vimtex/autoload/vimtex/complete.vim
162: ~/.config/nvim/plugged/vimtex/autoload/vimtex/debug.vim
163: ~/.config/nvim/plugged/vimtex/autoload/vimtex/doc.vim
164: ~/.config/nvim/plugged/vimtex/autoload/vimtex/echo.vim
165: ~/.config/nvim/plugged/vimtex/autoload/vimtex/env.vim
166: ~/.config/nvim/plugged/vimtex/autoload/vimtex/format.vim
167: ~/.config/nvim/plugged/vimtex/autoload/vimtex/fzf.vim
168: ~/.config/nvim/plugged/vimtex/autoload/vimtex/imaps.vim
169: ~/.config/nvim/plugged/vimtex/autoload/vimtex/include.vim
170: ~/.config/nvim/plugged/vimtex/autoload/vimtex/info.vim
171: ~/.config/nvim/plugged/vimtex/autoload/vimtex/jobs.vim
172: ~/.config/nvim/plugged/vimtex/autoload/vimtex/kpsewhich.vim
173: ~/.config/nvim/plugged/vimtex/autoload/vimtex/log.vim
174: ~/.config/nvim/plugged/vimtex/autoload/vimtex/matchparen.vim
175: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax.vim
176: ~/.config/nvim/plugged/vimtex/autoload/vimtex/misc.vim
177: ~/.config/nvim/plugged/vimtex/autoload/vimtex/motion.vim
178: ~/.config/nvim/plugged/vimtex/autoload/vimtex/pos.vim
179: ~/.config/nvim/plugged/vimtex/autoload/vimtex/profile.vim
180: ~/.config/nvim/plugged/vimtex/autoload/vimtex/scratch.vim
181: ~/.config/nvim/plugged/vimtex/autoload/vimtex/text_obj.vim
182: ~/.config/nvim/plugged/vimtex/autoload/vimtex/ui.vim
183: ~/.config/nvim/plugged/vimtex/autoload/vimtex/text_obj/targets.vim
184: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/packages.vim
185: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/p/amsmath.vim
186: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/p/siunitx.vim
187: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/p/cleveref.vim
188: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/p/booktabs.vim
189: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/p/url.vim
190: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/p/hyperref.vim
191: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/p/nameref.vim
192: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/p/array.vim
193: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/p/biblatex.vim
194: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/p/todonotes.vim
195: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/p/geometry.vim
196: ~/.config/nvim/plugged/vimtex/autoload/vimtex/syntax/p/tikz.vim
197: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/ftplugin/tex.vim
198: ~/.config/nvim/plugged/gruvbox-material/after/ftplugin/tex/gruvbox_material.vim
199: ~/.config/nvim/plugged/vimtex/after/ftplugin/tex.vim
200: ~/.config/nvim/after/ftplugin/tex.vim
201: ~/.config/nvim/plugged/vim-doge/autoload/doge.vim
202: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/autoload/provider/clipboard.vim
203: ~/.config/nvim/plugged/coc.nvim/autoload/coc/float.vim
204: ~/.config/nvim/plugged/coc.nvim/autoload/coc/cursor.vim
205: ~/.config/nvim/plugged/telescope.nvim/ftplugin/TelescopePrompt.lua
206: ~/.config/nvim/plugged/telescope.nvim/ftplugin/TelescopeResults.lua
207: ~/.config/nvim/plugged/ultisnips/autoload/UltiSnips.vim
208: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/autoload/provider/python3.vim
209: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/autoload/provider/pythonx.vim
210: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/autoload/remote/host.vim
211: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/autoload/provider.vim
212: ~/.config/nvim/plugged/vim-doge/autoload/doge/comment.vim
213: ~/.config/nvim/plugged/coc.nvim/autoload/coc/window.vim
214: ~/.config/nvim/plugged/coc.nvim/autoload/coc.vim
215: ~/.config/nvim/plugged/nvim-treesitter/autoload/nvim_treesitter.vim
216: ~/.config/nvim/plugged/vimtex/autoload/vimtex/parser/auxiliary.vim
217: ~/.config/nvim/plugged/vimtex/autoload/vimtex/jobs/neovim.vim
218: ~/.config/nvim/plugged/vimtex/autoload/vimtex/qf/u.vim
219: ~/.config/nvim/plugged/vimtex/autoload/vimtex/qf/biblatex.vim
220: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/syntax/qf.vim
221: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/ftplugin/qf.vim
222: ~/.config/nvim/plugged/nvim-bqf/after/ftplugin/qf/bqf.vim
223: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/syntax/pdf.vim
224: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/syntax/xml.vim
225: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/syntax/dtd.vim
226: ~/.config/nvim/plugged/vim-fugitive/autoload/fugitive.vim
227: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/scripts.vim
228: ~/.config/nvim/plugged/vim-fugitive/syntax/fugitive.vim
229: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/syntax/diff.vim
230: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/syntax/gitcommit.vim
231: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/ftplugin/gitcommit.vim
232: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/ftplugin/git.vim
233: ~/.config/nvim/plugged/gruvbox-material/after/ftplugin/gitcommit/gruvbox_material.vim
234: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/indent/json.vim
235: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/syntax/json.vim
236: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/ftplugin/json.vim
237: ~/.config/nvim/plugged/gruvbox-material/after/ftplugin/json/gruvbox_material.vim
238: ~/.config/nvim/after/ftplugin/json.vim
239: ~/.config/nvim/plugged/coc.nvim/autoload/coc/highlight.vim
240: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/syntax/help.vim
241: /nix/store/868a5yxbczsybalq2kcacdcx5pnkgar5-neovim-unwrapped-0.6.0/share/nvim/runtime/ftplugin/help.vim
242: ~/.config/nvim/plugged/gruvbox-material/after/ftplugin/help/gruvbox_material.vim
```

</p>
</details>

Here the variable setting works if I put it in line ` 16: ~/.config/nvim/core/plug.vim`, but has no effect if I put it in line ` 70: ~/.config/nvim/plugin/ultisnips.vim`.

**Comment by SirVer** [comment]
I do not understand what is going on in your case, it seems to me that the loading order is quite bugged. The relevant events as I can see it are these:

~~~
16: ~/.config/nvim/core/plug.vim
24: ~/.config/nvim/plugged/ultisnips/ftdetect/snippets.vim
70: ~/.config/nvim/plugin/ultisnips.vim
79: ~/.config/nvim/plugged/ultisnips/plugin/UltiSnips.vim
80: ~/.config/nvim/plugged/ultisnips/autoload/UltiSnips/map_keys.vim
116: ~/.config/nvim/plugged/ultisnips/after/plugin/UltiSnips_after.vim
207: ~/.config/nvim/plugged/ultisnips/autoload/UltiSnips.vim
~~~

The `UltiSnips_Manager` is supposed to be initialized in 207 [here](https://github.com/SirVer/ultisnips/blob/master/autoload/UltiSnips.vim#L8), this is were the setting to use or not use SnipMate snippets is baked in - after that changing the variable should not have any effect. However, changing it before that should be safe and no issue at all. 

Your data seems to indicate that the `UltiSnips_Manager` is initialized after 16, but before 70. However 24 is not containing any logic at all. It is not clear to me why you see the behaviour you do see.

Are you using UltiSnips python functions in any files in 16-69? If not, can you try to make your example more minimal and reduce the number of scripts that are executed?

**Comment by g6ai** [comment]
I have created an absolute minimal config, with only Ultisnips installed. And I tried to reproduce the problem by placing the variable in a `plugin/test.vim` file in the Neovim user rtp. The variable works well.

So this confirms the problem is not with UltiSnips, but somewhere in my own config. I'll use your advice to try to find the culprit somewhere in line 16-69, thank you very much for your info and time!

As this is not an issue of UltiSnips, I should close the issue here :)
